src/main/java/com/team4/testingsystem/services/TestsService.java

```diff
@@ -0,0 +1,94 @@
+package com.team4.testingsystem.services;
+
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.enums.Status;
+import com.team4.testingsystem.exceptions.TestNotFoundException;
+import com.team4.testingsystem.exceptions.UserNotFoundException;
+import com.team4.testingsystem.repositories.TestsRepository;
+import com.team4.testingsystem.repositories.UsersRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+public class TestsService {
+
+    private TestsRepository testsRepository;
+
+    private UsersRepository usersRepository;
+
+
+    @Autowired
+    public TestsService(TestsRepository testsRepository, UsersRepository usersRepository) {
+        this.testsRepository = testsRepository;
+        this.usersRepository = usersRepository;
+    }
+
+
+    public Iterable<Test> getAll() {
+        return testsRepository.findAll();
+    }
+
+    public Test getById(long id) {
+        try {
+            return testsRepository.findById(id).get();
+        } catch (Exception e) {
+            throw new TestNotFoundException();
+        }
+    }
+
+    public long createForUser(long userId) {
+
+       try {
+           Test test = Test.newBuilder()
+                   .setUser(usersRepository.findById(userId).get())
```

Comment:
Again, it can be written as follows:
```
User user = usersRepository.findById(userId).orElseThrow(UserNotFoundException::new);
```
And then:
```
.setUser(user)
```

Reply:
Not "can be" but "should be". Here we may set a null